src/catalog-debug/src/main.rs

```diff
@@ -135,7 +135,7 @@ async fn main() {
         enable_version_flag: true,
     });
     if let Err(err) = run(args).await {
-        eprintln!("catalog: fatal: {}", err.display_with_causes());
+        eprintln!("catalog: fatal: {:?}", err.display_with_causes());
```

Comment:
I _think_ the `display_with_causes` doesn't get you anything with the debug formatting, but also seems like it doesn't hurt, so fine to leave this as-is if you like https://github.com/MaterializeInc/materialize/blob/0ecbd3d54b3ec263752ec1b2d27155949959e90d/src/ore/src/error.rs#L80-L84

Reply:
Ah, sorry if I wasn't clear! From my read of the code I linked, I think it would be the same

Reply:
It's not actually clear to me that using the `Debug` format here is any better than `Display`, it could actually be worse.

The `Display` formatter iterates down a list of causes, meanwhile the `Debug` formatter just formats the underlying error in it's `Debug` format. Formatting the underlying error in `Debug` probably gets you more information, but I don't think it's guaranteed

Reply:
@ParkMyCar that underlying error is always an `anyhow::Error`, no? Here's the docs on how the various formatters work. To my eye, the Debug is a superset this mz_ore helper that prints the causes

https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations

Reply:
That makes sense. I reverted the debug and added a line to add the backtrace. What do you think about that?